workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,460 @@
+          dotnet publish -c Release -o ./published_app --no-restore
+
+      - name: Zip the published site for faster deployment and uploads
+        working-directory: ${{env.PROJECT_ROOT}}
+        # TODO:  If you plan on uploading artifacts, add this line to the run:
+        # (cd published_app && zip -r ../${{env.ARTIFACT_ZIP}} . -x */appsettings*.json appsettings*.json)
+        run: |
+          (cd published_app && zip -r ../${{env.DEPLOY_ZIP}} .)
+
+      - name: Stop App Pool
+        id: app-pool-stop
+        if: always()
+        runs-on: [self-hosted, windows-2019]
+        uses: im-open/app-pool-action@v1.0.0
+        with:
+          action: 'stop'
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Deploy deployment package
+        if: always() && steps.iis-stop.outcome == 'success'
+        uses: 'im-open/deploy-on-prem-web-app@v1.0.0'
+        with:
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          source-zip-file-path: ../${{env.DEPLOY_ZIP }}
+          deployment-folder-path: ${{ needs.set-vars.outputs.WEBSITE_PATH }}
+          clean-deployment-folder: 'true'
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Start App Pool
+        if: always() && steps.app-pool-stop.outcome == 'success'
+        runs-on: [self-hosted, windows-2019]
+        uses: im-open/app-pool-action@v1.0.0
+        with:
+          action: 'start'
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Close the PagerDuty Maintenance Window
+        if: always() && steps.open-window.outcome == 'success'
+        uses: im-open/close-pagerduty-maintenance-window@v1.0.0
+        with:
+          pagerduty-api-key: ${{ secrets.PAGERDUTY_API_KEY }}
+          maintenance-window-id: ${{ steps.open-window.outputs.maintenance-window-id }}
+
+      - name: Delete .zip and appsettings file that contains sensitive info
+        continue-on-error: true
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          rm -f ${{env.DEPLOY_ZIP}}
+          rm -f published_app
```

We learned this one with account management, this will fail because it's a folder.
```suggestion
          rm -rf published_app
```